firebase-config/src/main/java/com/google/firebase/remoteconfig/CustomSignals.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.remoteconfig;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper class which handles the storage and conversion to strings of key/value pairs with
+ * heterogeneous value types for custom signals.
+ */
+public class CustomSignals {
```

Comment:
Have we confirmed is this work on Kotlin as well? Looking into the amend public api review we have realized it is written in Java-esque Kotlin code.
Kotlin syntax should be like:
```
val customSignals = customSignals {
    put("userId", "a3974ec26c04")
    put("supportLevel", "premier")
    put("someIntSignal", 100L)
    put("someDoubleSignal", 1.23)
    put("key3", null)
  }
```
As @karenyz mentioned, let's add some unit tests for this data object as well. Also we should test either in unit test or test app to check kotlin syntax works as well

Reply:
Hi, I have tested the Kotlin syntax in a sample test app, and it works well. I've attached a screenshot for your reference.
<img width="1418" alt="Screenshot 2024-11-23 at 12 36 52 AM" src="https://github.com/user-attachments/assets/634872df-a344-4312-8388-a5ff3d7a33af">

Reply:
Thank you for the screen shot! Can we also update the public api review for kotlin to written in pure kotlin style?

Reply:
I see two potential ways to do this nicely in Kotlin.

1. Make a Kotlin function like:
   ```kotlin
   fun customSignals(builder: CustomSignals.Builder.() -> Unit) = CustomSignals.Builder().apply(builder).build()
   ```
   
   This will be easy and let the syntax will look like what @themiswang said.

2. Make a Kotlin extension like:
   ```kotlin
   fun FirebaseRemoteConfig.setCustomSignals(builder: CustomSignals.Builder.() -> Unit) = setCustomSignals(CustomSignals.Builder().apply(builder).build())
   ```
   
   This will let the syntax be more succinct and look like:
   ```kotlin
   Firebase.remoteConfig.setCustomSignals {
      put("userId", "a3974ec26c04")
      put("supportLevel", "premier")
      put("someIntSignal", 100L)
      put("someDoubleSignal", 1.23)
      put("key3", null)
   }
   ```

Either or both of these can live in the existing `RemoteConfig.kt` file.

Reply:
I have implemented it with Kotlin function and added tests for it as well.